pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -121,6 +121,9 @@ let
     versionOlder
   ;
 
+  # mixes libc and libxcrypt headers and libs and causes segfaults on importing crypt
+  libxcrypt = if stdenv.hostPlatform.isFreeBSD then null else inputs.libxcrypt;
```

Comment:
If this is an bona fide optional dependency, it would be more idiomatic to make a `withLibxcrypt ? libxcrypt != null && !stdenv.hostPlatform.isFreeBSD` flag, and change the logic to refer to that.

Reply:
This is not a bona fide optional dependency - freebsd libc provides xcrypt, and so it's just not needed on this platform. I don't think we can replace libxcrypt with libc at top-level, just because pretty much every (?) other package seems to want or at least be okay with the separate libxcrypt.

Reply:
OK, fair enough!